### PR TITLE
Handle deprecated messages caused by tf.nn.softmax_cross_entropy_with_logits

### PR DESCRIPTION
**Pull request by WindQAQ**
#397 
Use `tf.nn.softmax_cross_entropy_with_logits_v2` instead of `tf.nn.softmax_cross_entropy_with_logits`, which causes deprecated warnings (e.g. when running `cleverhans_tutorials/mnist_tutorial_tf.py`) for Tensorflow 1.8.

### CODE REVIEW

File: cleverhans/compat.py [hunk 1]
```diff
@@ -23,9 +23,9 @@ def reduce_function(op_func, input_tensor, axis=None, keepdims=None,
 
     if LooseVersion(tf.__version__) < LooseVersion('1.8.0'):
         warning = "Running on tensorflow version " + \
-                   LooseVersion(tf.__version__).vstring + \
-                   ". This version will not be supported by CleverHans" + \
-                   "in the future."
+            LooseVersion(tf.__version__).vstring + \
+            ". This version will not be supported by CleverHans" + \
+            "in the future."
```
Comment (goodfeli) [inline cleverhans/compat.py]: Can we change this to "Support for this version in CleverHans is deprecated and may be removed on or after 2019-01-26"?

File: cleverhans/compat.py [hunk 2]
```diff
@@ -85,3 +85,35 @@ def reduce_any(input_tensor, axis=None, keepdims=None,
...
+        raise ValueError("Both labels and logits must be provided.")
+
+    try:
+        labels = tf.stop_gradient(labels)
+        loss = tf.nn.softmax_cross_entropy_with_logits_v2(
+            labels=labels, logits=logits, dim=dim)
+    except AttributeError:
+        warning = "Running on tensorflow version " + \
+            LooseVersion(tf.__version__).vstring + \
+            ". This version will not be supported by CleverHans" + \
```
Comment (goodfeli) [inline cleverhans/compat.py]: same message here as above

**Comment by goodfeli** [comment]
Thanks! This looks good. Let's just put a specific date on the deprecation warning.

**Comment by WindQAQ** [comment]
Already added the date into warnings. Please check it!

**Pull request merged**